lgc/state/LgcContext.cpp

```diff
@@ -175,6 +175,47 @@ void LgcContext::initialize() {
 #endif
 }
 
+// =====================================================================================================================
+// Gets the name string of GPU target according to graphics IP version info.
+//
+// @param major : major version
+// @param minor : minor version
+// @param stepping : stepping info
+// @returns : LLVM GPU name as a std::string
+std::string LgcContext::getGpuNameString(unsigned major, unsigned minor, unsigned stepping) {
+  // A GfxIpVersion from PAL is three decimal numbers for major, minor and stepping. This function
+  // converts that to an LLVM target name, whith is "gfx" followed by the three decimal numbers with
+  // no separators, e.g. "gfx1010" for 10.1.0. A high stepping number 0xFFFA..0xFFFF denotes an
+  // experimental target, and that is represented by the final hexadecimal digit, e.g. "gfx101A"
+  // for 10.1.0xFFFA. In gfx9, stepping numbers 10..35 are represented by lower case letter 'a'..'z'.
+  std::string gpuName;
+  raw_string_ostream gpuNameStream(gpuName);
+  gpuNameStream << "gfx" << major << minor;
+  if (stepping >= 0xFFFA)
+    gpuNameStream << char(stepping - 0xFFFA + 'A');
+  else if (major == 9 && stepping >= 10)
+    gpuNameStream << char(stepping - 10 + 'a');
+  else
+    gpuNameStream << stepping;
+
+  return gpuNameStream.str();
+}
+
+static bool setTargetInfo(llvm::StringRef gpuName, TargetInfo *targetInfo) {
```

Comment:
This new function should have a function header comment.

Alternatively, the function seems a bit pointless to me. You could manually inline it into its two use sites, which returns one of them back to the state it was in before.

Reply:
Ok - I removed the function.